modelling/src/main/java/org/axonframework/modelling/SimpleRepository.java

```diff
@@ -0,0 +1,195 @@
+        var managedEntities = context.computeResourceIfAbsent(managedEntitiesKey, ConcurrentHashMap::new);
+
+        return managedEntities.computeIfAbsent(
+                entity.identifier(),
+                id -> {
+                    SimpleEntity<I, T> simpleEntity = SimpleEntity.mapToSimpleEntity(entity);
+                    attachEntitySaveToContext(id, simpleEntity, context);
+                    return CompletableFuture.completedFuture(simpleEntity);
+                }
+        ).resultNow();
+    }
+
+    @javax.annotation.Nonnull
+    @Override
+    public Class<T> entityType() {
+        return entityType;
+    }
+
+    @javax.annotation.Nonnull
+    @Override
+    public Class<I> idType() {
+        return idType;
+    }
+
+    @Override
+    public CompletableFuture<ManagedEntity<I, T>> load(@Nonnull I id, @Nonnull ProcessingContext context) {
+        var managedEntities = context.computeResourceIfAbsent(managedEntitiesKey, ConcurrentHashMap::new);
+
+        return managedEntities.computeIfAbsent(
+                id,
+                identifier -> loader
+                        .load(identifier, context)
+                        .thenApply((entity) -> {
+                            SimpleEntity<I, T> managedEntity = new SimpleEntity<>(id, entity);
+                            attachEntitySaveToContext(id, managedEntity, context);
+                            return managedEntity;
+                        })
+        );
+    }
+
+    private void attachEntitySaveToContext(I id, ManagedEntity<I, T> managedEntity, ProcessingContext context) {
+        context.onPrepareCommit(uow -> persister.persist(id, managedEntity.entity(), uow));
+    }
+
+    @Override
+    public CompletableFuture<ManagedEntity<I, T>> loadOrCreate(@Nonnull I identifier,
+                                                               @Nonnull ProcessingContext processingContext) {
+        return load(identifier, processingContext);
+    }
+
+    @Override
+    public ManagedEntity<I, T> persist(@Nonnull I id, @Nonnull T entity, @Nonnull ProcessingContext context) {
+        var managedEntities = context.computeResourceIfAbsent(managedEntitiesKey, ConcurrentHashMap::new);
+        if (managedEntities.containsKey(id)) {
+            ManagedEntity<I, T> managedEntity = managedEntities.get(id).join();
+            managedEntity.applyStateChange(oldState -> entity);
+            return managedEntity;
+        } else {
+            var managedEntity = new SimpleEntity<>(id, entity);
+            attachEntitySaveToContext(id, managedEntity, context);
+            managedEntities.put(id, CompletableFuture.completedFuture(managedEntity));
+            return managedEntity;
+        }
+    }
+
+    @Override
+    public void describeTo(@Nonnull ComponentDescriptor descriptor) {
+        descriptor.describeProperty("idClass", idType);
+        descriptor.describeProperty("entityClass", entityType);
```

Comment:
Shouldn't we describe the loader and persister too? What if those are implementation with their own fields?

Reply:
Yes, you are right. Adding.